app/assets/javascripts/modules/inline-image-modal.js

```diff
@@ -57,103 +58,62 @@ InlineImageModal.prototype.postModalForm = function (form) {
 
       return response.text()
         .then(function (text) {
-          return { body: text, finished: response.redirected }
+          return { body: text, done: response.redirected }
         })
     })
 }
 
+InlineImageModal.prototype.renderResponse = function (response) {
+  response
+    .then(function (result) {
+      this.$multiSectionViewer.showDynamicSection(result.body)
+      this.overrideActions()
+      this.initComponents()
```

Comment:
I wonder if you'd want to make these three things a single method so that this logic doesn't have to be repeated in metaInsert

Reply:
Indeed - I wanted to avoid too much indirection in this module, with too many methods. Most of the duplication is gone, but I'll try and address this when I do the refactoring PR to extract out generic functionality from this module.